### PR TITLE
chore: use firestore for referenceData

### LINKED ISSUE #2430: Bruke referanse-data fra Firestore
## Opprinnelse
Tidligere har referansedata og annen konfigurasjon ligget i Remote Config. Dette er uheldig siden det er en delay fra man oppdaterer Remote Config til det blir synlig app. I nettbutikken har vi allerede gått over til Firestore i denne saken: 
https://github.com/AtB-AS/webshop/issues/437

I saken nevnt ovenfor ble også noen andre konfigurasjoner overført til firestore. Man kan også gå over til å bruke de i appen.

## Analyse
preAssignedFareproducts_V2 ligger her: https://console.firebase.google.com/project/atb-mobility-platform-staging/firestore/data/~2Fconfiguration~2FreferenceData

På sikt vil vi flytte alt av konfigurasjon fra remoteConfig over til Firesstore.
Vi lager derfor en context i Appen, ConfigurationContext, der referansedata er en del av konfigurasjonen som ligger der.

## Test
Si fra Til Gunn så hun kan teste i QA

### PR DESCRIPTION
**Pull request by monopoint**
fixes #2430

### CODE REVIEW

File: src/screens/Ticketing/Tickets/use-recent-tickets.ts [hunk 1]
```diff
@@ -199,6 +202,7 @@ export default function useRecentTickets() {
   const [state, dispatch] = useReducer(reducer, initialState);
   const remoteConfigState = useRemoteConfig();
```
Comment (gorandalum) [inline src/screens/Ticketing/Tickets/use-recent-tickets.ts:203]: Denne er nok ikke nødvendig lenger etter at man har flyttet alle referanse-data.

File: src/configuration/FirestoreConfigurationContext.tsx [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+        (snapshot) => {
+          const preassignedFareproductsFromFirestore = snapshot.docs
+            .find((doc) => doc.id == 'referenceData')
+            ?.get<string>('preassignedFareProducts_v2');
+
+          const preassignedFareproducts = preassignedFareproductsFromFirestore
+            ? (JSON.parse(
+                preassignedFareproductsFromFirestore,
+              ) as PreassignedFareProduct[])
+            : defaultPreassignedFareProducts;
```
Comment (gorandalum) [inline src/configuration/FirestoreConfigurationContext.tsx:43]: Tror det er hakket ryddigere å sette inn `defaultPreassignedFareProducts` som default state i `useState`, også droppe denne logikken her i `onSnapshot`.

File: src/RemoteConfigContext.tsx [hunk 3]
```diff
@@ -117,10 +112,6 @@ const RemoteConfigContextProvider: React.FC = ({children}) => {
           config.modes_we_sell_tickets_for,
           defaultModesWeSellTicketsFor,
         ),
-        preassigned_fare_products: parseJson(
-          config.preassigned_fare_products,
-          defaultPreassignedFareProducts,
-        ),
         tariff_zones: parseJson(config.tariff_zones, defaultTariffZones),
         user_profiles: parseJson(config.user_profiles, defaultUserProfiles),
```
Comment (gorandalum) [inline src/RemoteConfigContext.tsx:116]: Disse er også referanse-data.

File: src/configuration/FirestoreConfigurationContext.tsx [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+  createContext,
+  useContext,
+  useEffect,
+  useMemo,
+  useState,
+} from 'react';
+import firestore from '@react-native-firebase/firestore';
+import {PreassignedFareProduct} from '@atb/reference-data/types';
+import Bugsnag from '@bugsnag/react-native';
+import {defaultPreassignedFareProducts} from '@atb/reference-data/defaults';
```
Comment (gorandalum) [inline src/configuration/FirestoreConfigurationContext.tsx:11]: Gjerne oppdater default-verdiene i koden med det som ligger inne i Firestore.

**Review by gorandalum — CHANGES_REQUESTED** [review]

File: src/screens/Ticketing/Tickets/use-recent-tickets.ts (hunk 1 above)
Comment (monopoint) [reply src/screens/Ticketing/Tickets/use-recent-tickets.ts]: fikset

File: src/configuration/FirestoreConfigurationContext.tsx (hunk 2 above)
Comment (monopoint) [reply src/configuration/FirestoreConfigurationContext.tsx]: 👍

File: src/configuration/FirestoreConfigurationContext.tsx (hunk 4 above)
Comment (monopoint) [reply src/configuration/FirestoreConfigurationContext.tsx]: filene er oppdatert

**Review by gorandalum — APPROVED** [review]

File: src/configuration/FirestoreConfigurationContext.tsx [hunk 5]
```diff
@@ -0,0 +1,161 @@
...
+    .find((doc) => doc.id == 'referenceData')
+    ?.get<string>('tariffZones');
+
+  try {
+    if (tariffZonesFromFirestore) {
+      return JSON.parse(tariffZonesFromFirestore) as TariffZone[];
+    }
+  } catch (error) {
+    Bugsnag.leaveBreadcrumb(
+      'Error parsing preassignedFareproducts from FireStore',
```
Comment (jorelosorio) [inline src/configuration/FirestoreConfigurationContext.tsx]: The message error should change to `Error parsing tariffZones from FireStore`

File: src/configuration/FirestoreConfigurationContext.tsx [hunk 6]
```diff
@@ -0,0 +1,161 @@
...
+    .find((doc) => doc.id == 'referenceData')
+    ?.get<string>('userProfiles');
+
+  try {
+    if (userProfilesFromFirestore) {
+      return JSON.parse(userProfilesFromFirestore) as UserProfile[];
+    }
+  } catch (error) {
+    Bugsnag.leaveBreadcrumb(
+      'Error parsing preassignedFareproducts from FireStore',
```
Comment (jorelosorio) [inline src/configuration/FirestoreConfigurationContext.tsx]: The message error should change to Error parsing userProfiles from FireStore

File: src/configuration/FirestoreConfigurationContext.tsx [hunk 7]
```diff
@@ -0,0 +1,161 @@
...
+
+  try {
+    if (preassignedFareproductsFromFirestore) {
+      return JSON.parse(
+        preassignedFareproductsFromFirestore,
+      ) as PreassignedFareProduct[];
+    }
+  } catch (error) {
+    Bugsnag.leaveBreadcrumb(
+      'Error parsing preassignedFareproducts from FireStore',
```
Comment (jorelosorio) [inline src/configuration/FirestoreConfigurationContext.tsx]: The message error should change to `Error parsing preassignedFareproducts in v2 from FireStore`? Not sure about this version 2 specification in the error message!

**Review by jorelosorio — COMMENTED** [review]
Hey! I came up with these small suggestions for your PR. Good work!

File: src/configuration/FirestoreConfigurationContext.tsx (hunk 7 above)
Comment (monopoint) [reply src/configuration/FirestoreConfigurationContext.tsx]: I don't think we actually need to version the data.
I'm not sure why data is named v2 in firestore, probably just for comparison/reference

Comment (gorandalum) [reply src/configuration/FirestoreConfigurationContext.tsx]: This is now changed to `Bugsnag.notify` and not as relevant. 

But regarding versioning: The versioning is actually important. We made some improvements to the reference data structure for `preassignedFareproducts` and made a v2, but old versions of the app are still fetching the old version (v1) of the reference data.

**Pull request merged**